lib/3scale_toolbox/commands/import_command/openapi/command.rb

```diff
@@ -0,0 +1,66 @@
+require 'swagger'
+require 'uri'
+require 'net/http'
+
+module ThreeScaleToolbox
+  module Commands
+    module ImportCommand
+      module OpenAPI
+        class OpenAPISubcommand < Cri::CommandRunner
+          include ThreeScaleToolbox::Command
+          include ThreeScaleToolbox::Remotes
+          include ResourceReader
+
+          def self.command
+            Cri::Command.define do
+              name        'openapi'
+              usage       'openapi [opts] -d <dst> <spec>'
+              summary     'Import API defintion in OpenAPI specification'
+              description 'Using an API definition format like OpenAPI, import to your 3Scale API'
+
+              option  :d, :destination, '3scale target instance. Format: "http[s]://<authentication>@3scale_domain"', argument: :required
+              option  :s, :service, '<service_id> of your 3Scale account', argument: :required
+              param   :openapi_resource
+
+              runner OpenAPISubcommand
+            end
+          end
+
+          def run
+            context = create_context
+            if options[:service]
+              context[:service] = Entities::Service.new(id: options[:service],
+                                                        remote: context[:threescale_client])
+            end
+
+            tasks = []
+            tasks << CreateServiceStep.new(context) unless options[:service]
```

Comment:
So, it is not possible to create and update the service by the same command?

Then eventually the pipeline importing the spec would have to know if the service exists or not and that can be hard.

Have you considered passing service system_name instead of the id, so it can be used to create/update the same service?

Reply:
Since the are no clear specs on the feature, this is another decision taken inspired from
https://github.com/3scale/3scale-cli. It is exactly the same behavior.

Even more, on the current implementation, since `system_name` is taken from swagger spec, trying to import twice the swagger definition on the same tenant (creating new services) will fail on the second try because `system_name` will be taken. It makes sense for me, so I implemented like that.

IMO the user should know whether new service is being created or a service is being updated (hence `service_id` needs to be known in advance). But if it has some value to abstract that decision from the user, providing `system_name` would indeed be the way to go.

Reply:
I think when used in pipeline (that is something we want to support) there is no way to know if you are creating or updating a service.
We would have to provde a command to get a service and then let customers write the logic.

IMO we should rather offer way for create or update out of the box to help automation.

Reply:
Good, then `--system-name` could be optional parameter. It not provided, `system_name` could be set from `info.title` from openAPI definiton.

Having the `system_name`, the toolbox will check whether service exists and run create/update accordingly. 

In fact, we could do the same with `update service` and `copy service`. The toolbox will, out of the box, perform required tasks from provided `system_name`. The two commands lead to confusion and they do the same task, one on newly created service and the other on existing one.

Reply:
So, even without supplying `--system-name` it would take it from the `info.title` and update the existing service, right?

Reply:
if the service would exist, then update. If not, create one. This behavior could be independent from the source of `system_name`

Reply:
👍  sounds perfect!